### PR TITLE
Windows executable symlink

### PR DESCRIPTION
**Pull request by gdavison**
Creates a symlink to the current Consul version at `C:\Program Files\consul\consul.exe`

Addresses #451

### CODE REVIEW

**Comment by legal90** [comment]
@gdavison,
Thank you for this Pull-Request! 👍 
As I see, `windows_path` is a pretty new resource, available only in Chef Client v13.4 and later.
Do you know - is there any other way to do the same on the earlier version of Chef Client?

**Comment by gdavison** [comment]
@legal90 The `windows_path` function was previously provided by the `windows` cookbook, then added to core Chef in v13.4. https://github.com/chef-cookbooks/windows#deprecated-resources-note

I'm going to update the cookbook code and explicitly include the `windows` cookbook as a dependency.

File: libraries/consul_installation_binary.rb [hunk 1]
```diff
@@ -59,6 +59,15 @@ def action_create
             to ::File.join(options[:extract_to], new_resource.version, 'consul')
             not_if { windows? }
           end
+
+          link "#{node.config_prefix_path}\\consul.exe" do
+            to ::File.join(options[:extract_to], new_resource.version, 'consul.exe')
+            only_if { windows? }
+          end
+          windows_path node.config_prefix_path do
```
Comment (legal90) [inline libraries/consul_installation_binary.rb]: Please, try if we can just use `env` instead:
```
  env 'path' do
    action :modify
    delim ::File::PATH_SEPARATOR
    value node.config_prefix_path
    only_if { windows? }
  end
```
It should work as expected, on Chef >= 12, without having to add a `windows` dependency

Comment (gdavison) [reply libraries/consul_installation_binary.rb]: I can certainly take a look.

But the `windows` cookbook isn't a new dependency: the `consul` cookbook depends on `build-essential`, which depends on `seven-zip`, which has a dependency on `windows`. So this is more making it an explicit dependency. And with `windows_path` becoming part of core Chef, it seems more clear going forward.

Let me know what you think.

Comment (legal90) [reply libraries/consul_installation_binary.rb]: > the `consul` cookbook depends on `build-essential`, which depends on `seven-zip`, which has a dependency on `windows`

Ah, sorry, I forgot about that. Then I'm OK with setting that dependency explicitly. 👍

Comment (gdavison) [reply libraries/consul_installation_binary.rb]: No worries, I had the "wait, why did that work" moment myself :)

**Pull request merged**